Connection/Connection.php

```diff
@@ -681,21 +683,31 @@ public function read($length, $flags = 0)
             );
         }
 
-        if (true === $this->isEncrypted()) {
-            return fread($this->getStream(), $length);
+        if (is_int($chunk)) {
```

Comment:
```diff
- if (is_int($chunk))
+ if (is_int($chunk) && 0 < $chunk))
```

to ensure the chunk is greater than zero. Could it be greater than or equal to zero?

Reply:
You right! never trust the userland!